### DISCUSSION TITLE
How long should data blobs be stored before they can be pruned?

### DISCUSSION THREAD

**Discussion by musalbas** [celestiaorg/celestia-node#2034] [General]
Currently, celestia-node storage nodes store all historical data blobs. However, the [purpose of data availability protocols is not store all historical data blobs for ever](https://notes.ethereum.org/@vbuterin/proto_danksharding_faq#If-data-is-deleted-after-30-days-how-would-users-access-older-blobs). Therefore, we should discuss and define a socially-accepted network-wide constant number of days ("**storage window**") after which the network does not guarantee that the data will be retrievable. We intend to make it possible for rollup full nodes to easily be able to contribute to the storage of historical data blobs within their rollup (see milestone 3 in https://github.com/celestiaorg/celestia-node/issues/2033).

**Comment by nashqueue** [comment]
The lower bound for this should be at least the Frauf Proof Window + Worst case Block Reconstruction time

**Reply by musalbas** [reply]
also + unbonding period too perhaps.

**Comment by Wondertan** [comment]
I would make it == unbonding period or trusting period.

**Reply by Wondertan** [reply]
This covers FP window and block reconstruction time.

**Reply by cmwaters** [reply]
Yeah I agree with having it equal to the unbonding period

**Reply by liamsi** [reply]
Same. I think unbonding period is sufficient. Though we can make it slightly more generous and cover e.g. the last 1-3 months instead of 3 weeks. Either way, the unbonding period is a reasonable lower bound.

**Comment by joshuacheong** [comment]
I am unclear about this in-production today, are the data blobs stored indefinitely today or are they pruned?

Personally I think with competitors like EIP-4844 danksharding, it is better if Celestia nodes hold the data for longer periods to be an archival back-up solution as well (after all the value of data availability sampling is supposed to enable Celestia to scale to the number of light nodes and can have much more capacity in-theory).

**Reply by musalbas** [reply]
>Personally I think with competitors like EIP-4844 danksharding, it is better if Celestia nodes hold the data for longer periods to be an archival back-up solution as well (after all the value of data availability sampling is supposed to enable Celestia to scale to the number of light nodes and can have much more capacity in-theory).

Unlike Ethereum nodes, Celestia does at least have a built-in data blob archival node, so that you can download historical data over the Celestia peer-to-peer network, as long as some archival nodes are serving the data, rather than having to use some third party protocol.

**Reply by renaynay** [reply]
>I am unclear about this in-production today, are the data blobs stored indefinitely today or are they pruned?


They are currently stored indefinitely, but eventually a pruning mode will be introduced that will prune data that is older than 30 days (the sampling window) defined in https://github.com/celestiaorg/CIPs/blob/main/cips/cip-4.md.

**Reply by renaynay** [reply]
Node runners will be able to specify if they would like to run a pruned node or an archival node.

**Reply by qzhodl** [reply]
Is the built-in BLOB archival node run by the Celestia team?

**Reply by musalbas** [reply]
No, it's a decentralized network

**Reply by qzhodl** [reply]
The trust assumption here is that there is at least one archival node in the network providing data to others. I am curious, is there any in-protocol incentive for these archival nodes?

**Reply by musalbas** [reply]
No, there isn't. Same as any other blockchain, there is no in protocol incentive for serving historical data on Bitcoin or Ethereum.
